src/util/permissionGroups.tsx

```diff
@@ -0,0 +1,187 @@
+import * as Yup from "yup";
+import { AbortControllerState, checkDuplicateName } from "./helpers";
+import { LxdGroup, LxdIdentity } from "types/permissions";
+import { IdentitiesOrGroupsChangeSummary } from "./permissionIdentities";
+
+export const testDuplicateGroupName = (
+  controllerState: AbortControllerState,
+  excludeName?: string,
+): [string, string, Yup.TestFunction<string | undefined, Yup.AnyObject>] => {
+  return [
+    "deduplicate",
+    "A group with this name already exists",
+    (value?: string) => {
+      return (
+        (excludeName && value === excludeName) ||
+        checkDuplicateName(value, "", controllerState, "auth/groups")
+      );
+    },
+  ];
+};
+
+export const getCurrentIdentitiesForGroups = (
+  groups: LxdGroup[],
+  identities: LxdIdentity[],
+) => {
+  const totalGroupsCount = groups.length;
+  const identityIdsInAllGroups: string[] = [];
+  const identityIdsInSomeGroups: string[] = [];
+  const identityIdsInNoGroups: string[] = [];
+  for (const identity of identities) {
+    let allocatedCount = 0;
+    const identityGroupsLookup = new Set(identity.groups || []);
+
+    for (const group of groups) {
+      if (identityGroupsLookup.has(group.name)) {
+        allocatedCount++;
+      }
+    }
+    const groupAllocatedToAll = allocatedCount === totalGroupsCount;
+    const groupAllocatedToSome = !groupAllocatedToAll && allocatedCount > 0;
+
+    if (groupAllocatedToAll) {
+      identityIdsInAllGroups.push(identity.id);
+      continue;
+    }
+
+    if (groupAllocatedToSome) {
+      identityIdsInSomeGroups.push(identity.id);
+      continue;
+    }
+
+    identityIdsInNoGroups.push(identity.id);
+  }
+
+  return {
+    identityIdsInAllGroups,
+    identityIdsInSomeGroups,
+    identityIdsInNoGroups,
+  };
+};
+
+export const getRelevantIdentities = (args: {
```

Comment:
Find a better name? Relevant is not transporting additional meaning.

Reply:
`addedOrRemovedIdentities`?

Reply:
Yes, sounds good.